lib/broadway/topology/rate_limiter.ex

```diff
@@ -16,20 +14,20 @@ defmodule Broadway.Topology.RateLimiter do
         name = Keyword.fetch!(opts, :name)
         producers_names = Keyword.fetch!(opts, :producers_names)
         args = {name, rate_limiting_opts, producers_names}
-        GenServer.start_link(__MODULE__, args, name: table_name(name))
+        GenServer.start_link(__MODULE__, args, name: rate_limiter_name(name))
     end
   end
 
-  def rate_limit(table_name, amount)
-      when is_atom(table_name) and is_integer(amount) and amount > 0 do
-    :ets.update_counter(table_name, @row_name, -amount)
+  def rate_limit(counter, amount)
+      when is_reference(counter) and is_integer(amount) and amount > 0 do
+    :atomics.sub_get(counter, 1, amount)
```

Comment:
Can we put the `1` index in a module attribute, similar to how we had `@row_name` before?

```elixir
@atomics_index 1
```

This way we know what this is and we'll always use the same index.